### PR TITLE
fix: Propagate rename to PDS members

### LINKED ISSUE #3314: Renaming a PDS does not update member nodes
**Describe the bug**

When I rename a PDS in Zowe Explorer, then opening its members fails because their resource URI is not updated.

**To Reproduce**

* Search for a dataset pattern that matches a PDS
* Expand the PDS node to reveal members
* Right-click the PDS node and select the Rename option
* After renaming the PDS, click on one of its members
* Notice the member fails to open because URI is outdated

**Expected behavior**

The resource URI stored on the member nodes should be updated when the PDS is renamed.

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- Zowe Explorer Version:
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Fixes #3314 

## Release Notes

Milestone: 3.0.3

Changelog:

- Fixed an issue where opening a PDS member after renaming an expanded PDS resulted in an error. [#3314](https://github.com/zowe/zowe-explorer-vscode/issues/3314)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by zFernand0 — APPROVED** [review]
I like that even though we are looping through the opened members, the text editor doesn't switch focus to each one of them (like it used to happen back in like v1.10.0 or something 😋 

TL;DR: LGTM! 😋

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks for the fix @traeok

**Review by JillieBeanSim — APPROVED** [review]
